src/devices/video/ef9345.h

```diff
@@ -31,6 +31,7 @@ class ef9345_device :   public device_t,
 public:
 	// construction/destruction
 	ef9345_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+	ef9345_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock, uint32_t variant);
```

Comment:
Pass-through constructors should be `protected`, not `public`.  Also, you need to pass through device type/name/shortname for it to work properly.